src/axom/primal/operators/detail/winding_number_impl.hpp

```diff
@@ -394,33 +434,40 @@ double stokes_winding_number_adaptive(const Point<T, 3>& query,
 
       // Compute one of three vector field line integrals depending on
       //  the orientation of the original surface, indicated through ax.
-      if(ax == SingularityAxis::x)
+      switch(ax)
+      {
+      case(SingularityAxis::x):
       {
         quad_fine[i] += quad_rule.IntPoint(q).weight *
           (node[2] * node[0] * node_dt[1] - node[1] * node[0] * node_dt[2]) /
           (node[1] * node[1] + node[2] * node[2]) / node_norm;
+        break;
       }
-      else if(ax == SingularityAxis::y)
+      case(SingularityAxis::y):
       {
         quad_fine[i] += quad_rule.IntPoint(q).weight *
           (node[0] * node[1] * node_dt[2] - node[2] * node[1] * node_dt[0]) /
           (node[0] * node[0] + node[2] * node[2]) / node_norm;
+        break;
       }
-      else  // ax == SingularityAxis::z || ax == SingularityAxis::rotated
+      case(SingularityAxis::z):
+      case(SingularityAxis::rotated):
       {
         quad_fine[i] += quad_rule.IntPoint(q).weight *
           (node[1] * node[2] * node_dt[0] - node[0] * node[2] * node_dt[1]) /
           (node[0] * node[0] + node[1] * node[1]) / node_norm;
+        break;
+      }
       }
     }
   }
 
-  constexpr int MAX_DEPTH = 12;
+  constexpr int MAX_DEPTH = 15;
```

Comment:
Should this parameter be exposed to users (with the current default value)?

Reply:
It certainly could, but the function call for this guy is already pretty long, between the three other kinds of error tolerances at play (`edge_tol`, `quad_tol`, and `EPS`). Furthermore, one of these, `quad_tol`, is already supposed to be the user-interpretable measure of accuracy, and the only reason there needs to be a cap on the number of times you recur in case `quad_tol` gets passed a zero.

For what it's worth, I didn't mean to update this value with the new bugfixes: `12` was the original minimum value you needed to set the cap to pass the tests, since even if you set the limit to something higher, you didn't need any more than 12. It's a fairly arbitrary cap, I'll admit, and I'm very open to discussion on how to improve it.